### PR TITLE
Bump to 0.3.0, fix Windows compilation and change license to BSD-3-Clause

### LINKED ISSUE #44: [windows-2019@Release@conda] CI failure

### LINKED ISSUE #45: Create a new release out of the current state
The latest release [v0.2.2](https://github.com/robotology/yarp-device-xsensmt/releases/tag/v0.2.2) is from November 2022, and many updates have been added since then.

There have been many cases where a fresh install via `robotology-superbuild` didn't work with some new devices, and the users had to switch to `master` in order to get it running, but it cost them some debugging time before figuring that out.

CC @gabrielenava @davidegorbani @GiulioRomualdi @G-Cervettini @traversaro

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/yarp-device-xsensmt/issues/45 .
Fix https://github.com/robotology/yarp-device-xsensmt/issues/44 .

For some reason the CI is super-slow, but that is another story.

### CODE REVIEW

**Comment by traversaro** [comment]
> For some reason the CI is super-slow, but that is another story.

Actually, I also fixed this in https://github.com/robotology/yarp-device-xsensmt/pull/46/commits/9062fc3ed7cf0bed4d3c20507c95e691a27f03f5, I could not wait for the CI to be happy.

**Comment by traversaro** [comment]
> > For some reason the CI is super-slow, but that is another story.
> 
> Actually, I also fixed this in [9062fc3](https://github.com/robotology/yarp-device-xsensmt/commit/9062fc3ed7cf0bed4d3c20507c95e691a27f03f5), I could not wait for the CI to be happy.

Cool, Linux CI time decreased from 23 minutes to 2 minutes, Windows from >= 60 minutes to 5 minutes.

**Comment by traversaro** [comment]
@HosameldinMohamed the PR is ready for review.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,5 @@
-# Copyright (C) 2017-2020 Fondazione Istituto Italiano di Tecnologia
-# Authors: Silvio Traversaro <silvio.traversaro@iit.it>
-# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LICENSE
+# SPDX-FileCopyrightText: Fondazione Istituto Italiano di Tecnologia (IIT)
+# SPDX-License-Identifier: BSD-3-Clause
 
 cmake_minimum_required(VERSION 3.5)
 project(yarp-device-xsensmt)
```
Comment (HosameldinMohamed) [inline CMakeLists.txt:5]: Regarding setting the project version in the CMake file, I see it's done for many other projects, but here it's not set.

I'm not sure if it's needed.

**Review by HosameldinMohamed — APPROVED** [review]
Great!

File: CMakeLists.txt (hunk 1 above)
Comment (traversaro) [reply CMakeLists.txt:5]: We are not installing any CMake version file for the project, so there is no benefit in keeping the version in the CMake project.

Comment (HosameldinMohamed) [reply CMakeLists.txt:5]: Makes sense!

**Pull request merged**